consensus/enclave/api/src/lib.rs

```diff
@@ -318,7 +326,7 @@ pub trait ConsensusEnclave: ReportableEnclave {
     fn form_block(
         &self,
         parent_block: &Block,
-        encrypted_txs_with_proofs: &[(WellFormedEncryptedTx, Vec<TxOutMembershipProof>)],
+        inputs: FormBlockInputs,
```

Comment:
Switching to this struct means that adding new transaction types doesn't involve changing a bunch of traits, messages, mocks, etc. Just the callers to `form_block` and the actual implementation.